conans/test/functional/subsystems_build_test.py

```diff
@@ -20,7 +25,129 @@ def test_cygwin_available(self):
         client.run_command('uname')
         assert "CYGWIN" in client.out
 
+    @pytest.mark.tool_msys2
+    @pytest.mark.tool_mingw32
+    def test_mingw32_available(self):
+        client = TestClient()
+        client.run_command('uname')
+        assert "MINGW32_NT" in client.out
+
+    @pytest.mark.tool_msys2
+    @pytest.mark.tool_mingw64
+    def test_mingw64_available(self):
+        client = TestClient()
+        client.run_command('uname')
+        assert "MINGW64_NT" in client.out
+
     def test_tool_not_available(self):
         client = TestClient()
         client.run_command('uname', assert_error=True)
         assert "'uname' is not recognized as an internal or external command" in client.out
+
+    makefile = textwrap.dedent("""
```

Comment:
Can be simplified to just the app, without library.

Reply:
simplified, now it build just one exe